software/chipwhisperer/capture/api/acquisition_controller.py

```diff
@@ -180,7 +180,7 @@ def doReadings(self, channelNumbers=None, tracesDestination=None, progressBar=No
         while self.currentTrace < self.maxtraces:
             if self.doSingleReading():
                 try:
-                    if self._writer:
+                    if self._scope and self._writer:
```

Comment:
The reason self._scope wasn't checked was intentional, so you could use this w/o actually having a scope (useful if you have an external scope recording data). But the use of the scope object is the actual bug. Will merge in then can add fix later (which means to pass a Null object as the trace I think, but will need to check that works).